nolearn/lasagne/visualize.py

```diff
@@ -338,29 +347,31 @@ def make_pydot_graph(layers, output_shape=True, verbose=False):
     return pydot_graph
 
 
-def draw_to_file(layers, filename, **kwargs):
+def draw_to_file(net, filename, **kwargs):
     """
     Draws a network diagram to a file
     :parameters:
-        - layers : list
-            List of the layers
+        - net : NeuralNet instance
+            The neural net to draw.
         - filename : string
             The filename to save output to
         - **kwargs: see docstring of make_pydot_graph for other options
     """
+    layers = net.get_all_layers()
     dot = make_pydot_graph(layers, **kwargs)
     ext = filename[filename.rfind('.') + 1:]
     with io.open(filename, 'wb') as fid:
         fid.write(dot.create(format=ext))
 
 
-def draw_to_notebook(layers, **kwargs):
+def draw_to_notebook(net, **kwargs):
```

Comment:
One thing I'm afraid of here is when (unknowingly) the user passes a net object which is not initialized. How could you handle this?